docs/grpc/c#.md

```diff
@@ -0,0 +1,217 @@
+
+`gRPC` is based on protocol buffers, and as such, you will need to compile the `lnd` proto file in C# before you can use it to communicate with `lnd`.
+
+This assumes you are using a Windows machine, but it applies equally to Mac and Linux.
+
+* Open Visual Studio, and create a new `.net core` console application called `lnrpc` at the root directory (Windows : `C:/`)
+
+* Within Visual Studio, open nuget package manager and install `Grpc.Tools` (1.17.0 at time of writing)
+
+* Open a `Cygwin` terminal and cd to your project folder:
+```bash
+cd C:/lnrpc/lnrpc
+```
+
+* Create the necessary folder structure, and then fetch the lnd [rpc.proto](https://github.com/lightningnetwork/lnd/blob/master/lnrpc/rpc.proto) file:
+```bash
+mkdir Grpc
+curl -o Grpc/rpc.proto -s https://raw.githubusercontent.com/lightningnetwork/lnd/master/lnrpc/rpc.proto
+```
+
+* Copy Google's [annotations.proto](https://github.com/googleapis/googleapis/blob/master/google/api/annotations.proto) to the correct folder:
+```bash
+mkdir Grpc/google
+mkdir Grpc/google/api
+curl -o Grpc/google/api/annotations.proto -s https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/annotations.proto
+```
+
+* Copy Google's [http.proto](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto) to the correct folder:
+```bash
+curl -o Grpc/google/api/http.proto -s https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/http.proto
+```
+
+* Copy Google's [descriptor.proto](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto) to the correct folder:
+```bash
+mkdir Grpc/google/protobuf
+curl -o Grpc/google/protobuf/descriptor.proto -s https://raw.githubusercontent.com/protocolbuffers/protobuf/master/src/google/protobuf/descriptor.proto
+```
+
+* Compile the proto file using `protoc.exe` from nuget package `Grpc.Tools` (remember to replace "YOUR_USER", and possibly version "1.17.0" in both paths):
+```bash
+# linux + mac nuget package location: ~/.nuget/packages
+cd Grpc
+C:/Users/<YOUR_USER>/.nuget/packages/grpc.tools/1.17.0/tools/windows_x64/protoc.exe --csharp_out . --grpc_out . rpc.proto --plugin=protoc-gen-grpc=C:/Users/<YOUR_USER>/.nuget/packages/grpc.tools/1.17.0/tools/windows_x64/grpc_csharp_plugin.exe
+```
+
+
+After following these steps, two files `Rpc.cs` and `RpcGrpc.cs` will be generated in the `Grpc` folder in your project.
+
+
+
+#### Imports and Client
+
+Every time you use C# `gRPC`, you will have to import the generated rpc classes, and use `nuget` package manger to install `Grpc.Core` (1.17.0 at time of writing), `Google.Protobuf` (3.6.1), and `Google.Api.CommonProtos` (1.4.0).
+
+After installing these, use the code below to set up a channel and client to connect to your `lnd` node:
+
+```c#
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Lnrpc;
```

Comment:
Also needed to do:
`dotnet add package Google.Protobuf --version 3.6.1`

Reply:
And `dotnet add package Grpc.Core --version 1.17.0`

Reply:
And `dotnet add package Google.Api.CommonProtos --version 1.4.0`

Don't know if all these separately downloade annotation protos are also somewhere in nuget?